lib/index.js

```diff
@@ -87,17 +87,19 @@ Mixpanel.prototype.loaded = function() {
 
 Mixpanel.prototype.page = function(page) {
   var category = page.category();
-  var name = page.fullName();
+  var name = page.name();
   var opts = this.options;
 
   // all pages
   if (opts.trackAllPages) {
-    this.track(page.track());
+    return this.track(page.track());
   }
 
   // categorized pages
-  if (category && opts.trackCategorizedPages) {
-    this.track(page.track(category));
+  if (opts.trackCategorizedPages && category) {
+    // If this option is checked and name was also passed, used the full name which includes both category & name
+    if (name) return this.track(page.track(page.fullName()));
+    return this.track(page.track(category));
```

Comment:
Just a style thing since this method shouldn't be returning a value, return on the next line makes that clear.